### ISSUE TITLE
`PD011`: False positive with non-dataframes

### ISSUE THREAD

**Issue by ngnpope** [charliermarsh/ruff#2480]
```python
class A:
    values: int

A().values
```
Results in:
```console
❯ ruff --version
ruff 0.0.239

❯ ruff --select PD011 bug.py 
bug.py:4:1: PD011 Use `.to_numpy()` instead of `.values`
Found 1 error.
```

**Comment by charliermarsh** [comment]
Similar to #2229. It's just really hard to avoid these without making the rules mostly useless. The best we could do is gate on the presence of `import pandas`, but even then you could have false positives, and you'd get more false negatives, since you don't have to import Pandas in a given file in order to work with DataFrames.

**Comment by JonathanPlasse** [comment]
Tell me if I am wrong, but could it be resolved if we tracked the types of variables?
Is it planned to support it?
For this would need first to have multiple files support.

**Comment by charliermarsh** [comment]
It could definitely be resolved if we had full type inference!

**Comment by ngnpope** [comment]
Yeah, it's a hard one without typing. I'll ignore for now. Sorry for the duplicate

**Comment by charliermarsh** [comment]
All good!
